cinema/views.py

```diff
@@ -31,9 +46,37 @@ class CinemaHallViewSet(viewsets.ModelViewSet):
 
 
 class MovieViewSet(viewsets.ModelViewSet):
-    queryset = Movie.objects.all()
+
+    @staticmethod
+    def _params_to_ints(string):
+        return [int(id) for id in string.split(",")]
+
+    queryset = Movie.objects.all().prefetch_related("actors", "genres")
     serializer_class = MovieSerializer
 
+    def get_queryset(self):
+        queryset = self.queryset
+        actors = self.request.query_params.get("actors")
+        genres = self.request.query_params.get("genres")
+        title = self.request.query_params.get("title")
+
+        if title:
+            queryset = queryset.filter(title__icontains=title)
+
+        if actors:
+            actors_ids = self._params_to_ints(actors)
+            queryset = queryset.filter(actors__id__in=actors_ids)
+
+        if genres:
+            genres_ids = self._params_to_ints(genres)
+            queryset = queryset.filter(genres__id__in=genres_ids)
+
+        if self.action in ("list", "retrieve"):
+            queryset = queryset.prefetch_related("actors")
+            queryset = queryset.prefetch_related("genres")
```

Comment:
you can set all models for prefetch_related in one queryset, dont need to make another one